xtask/src/zombienet.rs

```diff
@@ -0,0 +1,65 @@
+use duct::{cmd, IntoExecutablePath};
+
+pub struct Zombienet(duct::Handle);
+
+impl Zombienet {
+    // Try launching the network using zombienet.
+    // The 'out' argument lets you decide whether to
+    // show or not the standard output of the command
+    pub fn try_new(out: bool) -> anyhow::Result<Self> {
+        // binaries are expected to be in the path
+        //
+        // remove zombienet folder if it exists
+        // TODO: maybe is better to avoid deleting folder in scripts
+        cmd!("sh", "-c", "rm -r zombienet").unchecked().run()?;
+
+        // Check if the zombienet binary is available.
+        println!("check binaries availability ");
+
+        if let Err(_) = cmd!("sh", "-c", "command -v zombienet").run() {
+            return anyhow::bail!(
+                "'zombienet' is not found in PATH. Install zombienet. \n \
+             Available at https://github.com/paritytech/zombienet"
+            );
+        }
+
+        if let Err(_) = cmd!("sh", "-c", "command -v polkadot").run() {
+            return anyhow::bail!(
+            "'polkadot' is not found in PATH. Install polkadot \n \
+              To obtain, refer to https://github.com/paritytech/polkadot-sdk/tree/master/polkadot#polkadot"
+        );
+        }
+
+        if let Err(_) = cmd!("sh", "-c", "command -v sugondat-node").run() {
+            println!("");
+            return anyhow::bail!(
+            "'sugondat-node' is not found in PATH.  \n \
+             cd to 'sugondat/chain' and run 'cargo build --release' and add the result into your PATH."
+        );
+        }
+
+        println!("Launching zombienet");
+        let mut zombienet_cmd = cmd!(
+            "zombienet",
+            "spawn",
+            "-p",
+            "native",
+            "--dir",
+            "zombienet",
```

Comment:
I am wondering if we should be smarter about it in two regards:

1. control the ./zombienet folder. I.e. remove it before the launch by default.
2. use a directory ignored in .gitignore.

Also, another thought is that sovereign rollup demo may also be pointed to store its data in that directory.

Reply:
Zombienet is already in the .gitignore; otherwise, we would need to delete it every time we want to push something between tests. 

Currently, the script already deletes it if present. Another approach could be to delete it when the test is finished, but in my opinion, it is not good for two reasons: 

1. Sometimes, you want to see the logs after the test has executed. 
2. Zombienet will not return but will just hang if it finds an already existing zombienet folder, and it will not be visible in the test, with all the output mixed (zombienet, shim, sovereign).

Reply:
> Sometimes, you want to see the logs after the test has executed.

As I said, "remove it before the launch by default". So the user will have the ability to check the logs until the user runs the test again.

> Zombienet will not return but will just hang if it finds an already existing zombienet folder, and it will not be visible in the test, with all the output mixed (zombienet, shim, sovereign).

That's why I say it should be deleted before the launch.